middle_end/compilenv.ml

```diff
@@ -431,7 +428,8 @@ let structured_constants () =
   let provenance : Clambda.usymbol_provenance =
     { original_idents = [];
       module_path =
-        Path.Pident (Ident.create_persistent (current_unit_name ()));
+        Path.Pident (Ident.create_persistent (Compilation_unit.Name.to_string (
```

Comment:
We're not using this provenance info yet (it's for DWARF) but when we do, we'll actually want the full module path here, including any pack prefix.  So the conversion to Compilation_unit.Name should be removed, I think.

Reply:
Ah actually maybe it's easiest to leave a CR about this.  You would need something different from `Pident` in the pack case as a proper path would need constructing.

Reply:
Left a CR.